### PR TITLE
Add capacity to device config [HZ-870]

### PR DESCRIPTION
**Pull request by fbarotov**
EE https://github.com/hazelcast/hazelcast-enterprise/pull/4615

Resolves https://github.com/hazelcast/hazelcast/issues/20498
Resolves https://github.com/hazelcast/hazelcast-enterprise/issues/4578

### CODE REVIEW

File: hazelcast/src/main/resources/hazelcast-default.xml [hunk 1]
```diff
@@ -257,10 +257,10 @@
     <!--
         Configuration for a device, which a tiered-store can reference and use for its disk-tier.
     -->
-    <local-device name="default">
+    <local-device name="default-tiered-store-device">
```
Comment (blazember) [inline hazelcast/src/main/resources/hazelcast-default.xml:260]: Shouldn't we add the default capacity here as well?

File: hazelcast/src/main/java/com/hazelcast/config/DeviceConfig.java [hunk 2]
```diff
@@ -27,4 +29,9 @@
      * otherwise, returns {@code false} if this configuration is for a remote device.
      */
     boolean isLocal();
+
+    /**
+     * Returns the device capacity.
+     */
+    MemorySize getCapacity();
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/config/DeviceConfig.java:36]: I'd move the methods and fields from `MemorySize` to a new `Capacity` class and would make `MemorySize` extending `Capacity`, as well as marking it `@Deprecated`. We now use it not only for the native memory size, but for the hybrid log's memory and device tiers' capacity as well. Potentially in a follow-up PR. WDYT? cc @vbekiaris

**Review by blazember — CHANGES_REQUESTED** [review]
We need to include the spring config as well. Otherwise LGTM.

File: hazelcast/src/main/java/com/hazelcast/config/DeviceConfig.java (hunk 2 above)
Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/config/DeviceConfig.java:36]: @blazember good suggestion, +1

**Comment by blazember** [comment]
Also, this resolves https://github.com/hazelcast/hazelcast-enterprise/issues/4578 as well, doesn't it?

**Comment by fbarotov** [comment]
> Also, this resolves hazelcast/hazelcast-enterprise#4578 as well, doesn't it?

Yes, it does, thanks for the note :)

File: hazelcast/src/main/resources/hazelcast-default.xml (hunk 1 above)
Comment (fbarotov) [reply hazelcast/src/main/resources/hazelcast-default.xml:260]: Thanks for the catch, yes we should. Addressed in https://github.com/hazelcast/hazelcast/pull/20523/commits/b9cff34b927f7276da3075d72016d7afabc5a7df

File: hazelcast/src/main/java/com/hazelcast/config/DeviceConfig.java (hunk 2 above)
Comment (fbarotov) [reply hazelcast/src/main/java/com/hazelcast/config/DeviceConfig.java]: Thanks for the suggestion, addressed in https://github.com/hazelcast/hazelcast/pull/20523/commits/d80acd7191b204815139ceed66af50f14a0539f0

**Comment by fbarotov** [comment]
run-lab-run

**Comment by fbarotov** [comment]
Regarding changes to `MemorySize , Capacity, MemoryUnit` we discussed it with Zoltan (thank you for your suggestions :) ) in Slack as well. I can work on it in a separate PR, although partial modifications are in this PR. Having compactor update PR merged earlier would allow us to observe its behaviour for longer. wdyt? @blazember

**Comment by blazember** [comment]
I agree, naming is a separate concern. Let's merge the improvement and do the bikeshedding in a separate PR :)

**Review by blazember — APPROVED** [review]

**Comment by fbarotov** [comment]
Thank you everyone for the reviews :)

**Pull request merged**